kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/DeltaTableClusteringSuite.scala

```diff
@@ -182,20 +182,55 @@ class DeltaTableClusteringSuite extends DeltaTableWriteSuiteBase {
     }
   }
 
-  test("update a table with clustering columns should be blocked") {
+  test("update a non-clustered table with clustering columns should succeed") {
     withTempDirAndEngine { (tablePath, engine) =>
       createEmptyTable(engine, tablePath, testPartitionSchema)
       val table = Table.forPath(engine, tablePath)
-      val txnBuilder = table.createTransactionBuilder(engine, testEngineInfo, WRITE)
+      updateTableMetadata(engine, tablePath, clusteringColsOpt = Some(testClusteringColumns))
+      // Verify the clustering feature is included in the protocol
+      val snapshot = table.getLatestSnapshot(engine).asInstanceOf[SnapshotImpl]
+      // Verify the clustering domain metadata is written
+      verifyClusteringDomainMetadata(snapshot)
+    }
+  }
```

Comment:
please add tests for:
1) updating an existing table with clustering columns that doesn't exist in the table
2) Updating a clustered table with new clustering columns that
     i) subset of the original clustering columns
     ii) remove existing column and add a new one.
4) insert data with after updates.